lib/semantic_puppet/dependency.rb

```diff
@@ -120,15 +120,15 @@ def walk(graph, dependencies, *considering)
       name = dependencies.keys.sort.first
       deps = dependencies.delete(name)
 
+      @module_dependencies |= [name]
+
```

Comment:
Just curious, do you have a real-life example of this failing?

Outside of tests I'm not able to replicate this. I expected this to impact https://tickets.puppetlabs.com/browse/PUP-11172 in a way but I'm getting the same failures both with and without your PR. I believe the change makes sense though.

Reply:
I ran into it this way:
https://gist.github.com/ekohl/055633dfd53e69af8342bf325a1771eb#file-check_puppet_dependencies-rb-L95-L101